spec/index.bs

```diff
@@ -788,15 +786,20 @@ dictionary GPULimits {
     u32 maxBindGroups = 4;
 };
 
+interface GPUPendingBuffer {
+    readonly attribute ArrayBuffer buffer;
+    GPUBuffer finish();
+};
+
 // Device
 interface GPUDevice : GPUObjectBase {
     readonly attribute GPUExtensions extensions;
     readonly attribute GPULimits limits;
     readonly attribute GPUAdapter adapter;
 
     GPUBuffer createBuffer(GPUBufferDescriptor descriptor);
-    GPUMappedBuffer createBufferMapped(GPUBufferDescriptor descriptor);
-    Promise<GPUMappedBuffer> createBufferMappedAsync(GPUBufferDescriptor descriptor);
+    GPUPendingBuffer createBufferWithData(GPUBufferDescriptor descriptor);
```

Comment:
Instead of returning an array buffer for the author to fill-in, why doesn't the author just provide the array buffer as an argument? This function is already a convenience on top of regular mapping, so why can't we make it more convenient?

Reply:
The ArrayBuffer still needs to be in ipc shared memory to minimize the number of copies. An app might not copy into it - it might decode/decompress something directly into it.